front/src/common/uploadFileModal.tsx

```diff
@@ -28,7 +30,12 @@ const UploadFileModal = ({ handleSubmit }: UploadFileModalProps) => {
       }
       return undefined;
     } catch (error) {
-      return t('errorMessages.errorInvalidXMLFormat').toString();
+      setFailure(
+        castErrorToFailure(error, {
+          message: t('errorMessages.errorInvalidXMLFormat'),
+        })
+      );
+      return undefined;
```

Comment:
The function always returns `undefined` so the `if (xmlError) {` line 62 is never entered

Reply:
I guess remove the 2 returns + the if l:62